api/omnitransaction.py

```diff
@@ -0,0 +1,124 @@
+            self.testnet =True
+            self.magicbyte = 111
+            self.exodus_address=self.testnet_exodus_address
+
+        try:
+          if config.D_PUBKEY and ( 'donate' in form ) and ( form['donate'] in ['true', 'True'] ):
+            print "We're Donating to pubkey for: "+pybitcointools.pubkey_to_address(config.D_PUBKEY)
+            self.pubkey = config.D_PUBKEY
+          else:
+            print "not donating"
+            self.pubkey = form['pubkey']
+        except NameError, e:
+          print e
+          self.pubkey = form['pubkey']
+        self.fee = estimateFee(self.confirm_target)['result']
+        self.rawdata = form
+        self.tx_type = tx_type
+
+    def get_unsigned(self):
+        # get payload
+        payload = self.__generate_payload()
+        # Add exodous output
+        rawtx = createrawtx_reference(self.exodus_address)['result']
+        if 'transaction_to' in self.rawdata:
+            # Add reference for reciever
+            rawtx = createrawtx_reference(self.rawdata['transaction_to'], rawtx)['result']
+
+        # Add the payload    
+        if len(payload) <= 160:  #80bytes
+            rawtx = createrawtx_opreturn(payload, rawtx)['result']
+        else:
+            rawtx = createrawtx_multisig(payload, rawtx)['result']
+
+        # Decode transaction to get total needed amount
+        decodedtx = decoderawtransaction(rawtx)['result']
+
+        # Sumup the outputs
+        fee_total = Decimal(self.fee)
+        for output in decodedtx['vout']:
+            fee_total += Decimal(output['value'])
+
+        fee_total_satoshi = int( round( fee_total * Decimal(1e8) ) )
+
+        # Get utxo to generate inputs
+        print "Calling bc_getutxo with ", self.rawdata['transaction_from'], fee_total_satoshi
+        dirty_txes = bc_getutxo( self.rawdata['transaction_from'], fee_total_satoshi )
+        print "received", dirty_txes
+
+        if (dirty_txes['error'][:3]=='Con'):
+            raise Exception({ "status": "NOT OK", "error": "Couldn't get list of unspent tx's. Response Code: " + dirty_txes['code']  })
+
+        if (dirty_txes['error'][:3]=='Low'):
+            raise Exception({ "status": "NOT OK", "error": "Not enough funds, try again. Needed: " + str(fee_total) + " but Have: " + dirty_txes['avail']  })
+
+        total_amount = dirty_txes['avail']
+        unspent_tx = dirty_txes['utxos']
+
+        change = total_amount - fee_total_satoshi
+
+        #DEBUG 
+        print [ "Debugging...", dirty_txes,"miner fee sats: ", self.fee, "change: ",change,"total_amt: ", total_amount,"fee tot sat: ", fee_total_satoshi,"utxo ",  unspent_tx,"to ", self.rawdata['transaction_to'] ]
+
+        #source script is needed to sign on the client credit grazcoin
+        hash160=bc_address_to_hash_160(self.rawdata['transaction_from']).encode('hex_codec')
+        prevout_script='OP_DUP OP_HASH160 ' + hash160 + ' OP_EQUALVERIFY OP_CHECKSIG'
```

Comment:
This is what we do on the frontend with that script (sourceScript), I think it is needed like that for tx signing, but that's code we've been carrying around since the begining I think. do you think it's is not needed anymore? 


                var bytes = Bitcoin.Util.hexToBytes(unsignedTransactionHex);
                var transaction = Bitcoin.Transaction.deserialize(bytes);
                var script = self.parseScript(sourceScript);

                if (transaction.ins.length == 0) {
                    return {
                        waiting: false,
                        transactionError: true,
                        error: 'Error: Not enough inputs in the address!'
                    };
                }
                transaction.ins.forEach(function(input) {
                    input.script = script;
                });
                return transaction;